pkg/controller/pxc/controller.go

```diff
@@ -348,6 +341,10 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(request reconcile.Request) (re
 		}
 
 		haProxyService := pxc.NewServiceHAProxy(o)
+		err = setControllerReference(o, haProxyService, r.scheme)
```

Comment:
why not set Controller Reference inside NewServiceHAProxy?

Reply:
f172e9c1bdc04e1157d11b686a989e59a228e8dd